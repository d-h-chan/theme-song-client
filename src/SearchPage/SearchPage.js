import React, { Component } from 'react';
import Context from '../ContextManagement/Context.js'
import DatabaseApiService from '../services/DatabaseApiService'

class SearchPage extends Component {

  static contextType = Context;

  createTableRow = (item, index) => {
    return (
      <tr key={index}>
        <td><a href={item.url} target="_blank">{item.title}</a></td>
        <td>{item.artist}</td>
      </tr>)
  }

  handleSubmit = (event) => {
    event.preventDefault()
    let searchParam = event.target.searchDB.value
    DatabaseApiService.getSongsFromDatabase(searchParam)
      .then(this.context.setSearchPageResults)
  }

  render() {
    return (
      <>
        <form onSubmit={this.handleSubmit}>
          <label htmlFor="searchDB">Search:</label>
          <input type="text" id="searchDB" name="searchDB"></input>
          <input type="submit"></input>
        </form>

        <table>
          <tbody>
            {this.context.searchPageResults.length > 0 && (
              <tr>
              <th>Title</th>
              <th>Artist</th>
            </tr>
            )}
            {this.context.searchPageResults.map(this.createTableRow)}
          </tbody>
        </table>
      </>
    );
  }
}

export default SearchPage;