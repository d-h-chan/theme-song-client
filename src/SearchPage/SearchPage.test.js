import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SearchPage from './SearchPage'

it('renders without errors', () => {
  const wrapper = shallow(<SearchPage />)
  expect(toJson(wrapper)).toMatchSnapshot()
})