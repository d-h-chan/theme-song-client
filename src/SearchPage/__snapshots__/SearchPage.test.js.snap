// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without errors 1`] = `
<Fragment>
  <h2>
    Search Database
  </h2>
  <form
    onSubmit={[Function]}
  >
    <label
      className="upperCase"
      htmlFor="searchDB"
    >
      Search:
    </label>
    <input
      id="searchDB"
      name="searchDB"
      placeholder="Example: 'love'"
      type="text"
    />
    <input
      type="submit"
      value="search"
    />
  </form>
  <hr />
  <table>
    <tbody />
  </table>
  <LoadingIndicator />
</Fragment>
`;
